sandbox/src/main/java/org/robolectric/internal/bytecode/ClassInstrumentor.java

```
@@ -407,6 +383,44 @@ protected void instrumentConstructor(MutableClass mutableClass, MethodNode metho
     mutableClass.addMethod(initMethodNode);
   }
 
+  /**
+   * JaCoCo instrumentation initializes its probes before constructor to collect the coverage.
+   *
+   * <p>Those JaCoCo instrumented instructions are supported to be stripped from call super
+   * instructions and insert to the original constructor in case of breaking the following
+   * instructions.
+   */
+  private InsnList extractInitJaCoCoFromConstructorCallSuper(InsnList callSuper) {
+    InsnList removedInstructions = new InsnList();
+
+    AbstractInsnNode[] insns = callSuper.toArray();
+    for (int i = 0; i < insns.length; i++) {
+      AbstractInsnNode node = insns[i];
+
+      if (node.getOpcode() == Opcodes.INVOKESTATIC) {
+        MethodInsnNode mnode = (MethodInsnNode) node;
+        if ("$jacocoInit".equals(mnode.name)) {
+          if (i + 1 >= insns.length) {
+            throw new RuntimeException("huh? JaCoCo instrumented instructions completed?!");
+          }
+
+          if (insns[i + 1].getOpcode() != Opcodes.ASTORE) {
+            throw new RuntimeException("huh? JaCoCo instrumented instructions no astore?!");
+          }
+
+          callSuper.remove(insns[i]);
+          callSuper.remove(insns[i + 1]);
+          removedInstructions.add(insns[i]);
+          removedInstructions.add(insns[i + 1]);
+
+          return removedInstructions;
+        }
+      }
+    }
+
+    throw new RuntimeException("huh? JaCoCo instrumented?!");
```

Is it fine to throw `RuntimeException` here?

---

That could be a violation between robolectric's JaCoCo detection and JaCoCo instructions strip and split when it comes to line 421.
I think it would be better to fail early to avoid the previous issue when we has assumed that all the JaCoCo detection and JaCoCo instructions strip and split are breaked.

---

But some users can accept no-code-coverage or other things, but they can't accept exceptions when running tests.